builtin/v9/migration/test/migration_test.go

```diff
@@ -0,0 +1,99 @@
+package migration
+
+import (
+	"context"
+	"testing"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/manifest"
+	"github.com/ipfs/go-cid"
+
+	"github.com/filecoin-project/go-state-types/builtin/v9/migration"
+
+	"github.com/filecoin-project/go-state-types/builtin"
+	system9 "github.com/filecoin-project/go-state-types/builtin/v9/system"
+	"github.com/filecoin-project/go-state-types/builtin/v9/util/adt"
+	cbor "github.com/ipfs/go-ipld-cbor"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNv16Migration(t *testing.T) {
+	ctx := context.Background()
+	bs := cbor.NewMemCborStore()
+	adtStore := adt.WrapStore(ctx, bs)
+
+	startRoot := makeInputTree(ctx, t, adtStore)
+
+	oldStateTree, err := migration.LoadTree(adtStore, startRoot)
+	require.NoError(t, err)
+
+	oldSystemActor, found, err := oldStateTree.GetActor(builtin.SystemActorAddr)
+	require.NoError(t, err)
+	require.True(t, found, "system actor not found")
+
+	var oldSystemState system9.State
+	err = adtStore.Get(ctx, oldSystemActor.Head, &oldSystemState)
+	require.NoError(t, err)
+
+	oldManifestDataCid := oldSystemState.BuiltinActors
+	oldManifest := manifest.Manifest{
+		Version: 1,
+		Data:    oldManifestDataCid,
+	}
+	require.NoError(t, oldManifest.Load(ctx, adtStore), "failed to load old manifest")
+
+	newManifestCid, newManifestDataCid := makeTestManifest(t, adtStore, "fil/9/")
+	log := migration.TestLogger{TB: t}
+
+	cache := migration.NewMemMigrationCache()
+	_, err = migration.MigrateStateTree(ctx, adtStore, newManifestCid, startRoot, 200, migration.Config{MaxWorkers: 1}, log, cache)
+	require.NoError(t, err)
+
+	cacheRoot, err := migration.MigrateStateTree(ctx, adtStore, newManifestCid, startRoot, 200, migration.Config{MaxWorkers: 1}, log, cache)
+	require.NoError(t, err)
+
+	noCacheRoot, err := migration.MigrateStateTree(ctx, adtStore, newManifestCid, startRoot, 200, migration.Config{MaxWorkers: 1}, log, migration.NewMemMigrationCache())
+	require.NoError(t, err)
+	require.True(t, cacheRoot.Equals(noCacheRoot))
+
+	// TODO: check all the CIDs
```

Comment:
Get rid of this TODO (in both versions of this test). We compare the CIDs on line 93

Reply:
What do you mean by "in both versions of this test"?

Reply:
This one and the one you copied it from.